utils/rawDisk.go

```diff
@@ -0,0 +1,188 @@
+package utils
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"encoding/binary"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+
+	"github.com/kairos-io/kairos-sdk/constants"
+	"github.com/kairos-io/kairos-sdk/types"
+)
+
+// Raw2Azure converts a raw disk to a VHD disk compatible with Azure
+// All VHDs on Azure must have a virtual size aligned to 1 MB (1024 × 1024 bytes)
+// The Hyper-V virtual hard disk (VHDX) format isn't supported in Azure, only fixed VHD
+func Raw2Azure(source string, logger types.KairosLogger) error {
```

Comment:
Is this implementing this script? https://github.com/kairos-io/AuroraBoot/blob/main/image-assets/azure.sh

Reply:
mainly yes, but in pure go instead.